cli/src/main.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+use color_eyre::eyre::{
+    self,
+    WrapErr,
+};
+use frame_metadata::RuntimeMetadataPrefixed;
+use scale::{
+    Decode,
+    Input,
+};
+use std::{
+    fs,
+    io::{
+        self,
+        Read,
+        Write,
+    },
+    path::PathBuf,
+};
+use structopt::StructOpt;
+
+/// Utilities for working with substrate metadata for subxt.
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(subcommand)]
+    command: Command,
+}
+
+#[derive(Debug, StructOpt)]
+enum Command {
+    /// Download metadata from a substrate node, for use with `subxt` codegen.
+    #[structopt(name = "metadata")]
+    Metadata {
+        /// the url of the substrate node to query for metadata
+        #[structopt(
+            name = "url",
+            long,
+            parse(try_from_str),
+            default_value = "http://localhost:9933"
+        )]
+        url: url::Url,
+        /// the format of the metadata to display: `json`, `hex` or `bytes`
+        #[structopt(long, short, default_value = "json")]
+        format: String,
```

Comment:
Nit: format could be an enum with a `FromStr` impl so that structopt can handle the error message for you if it's wrong?

Reply:
#309